src/core/agent/services/agentService.ts

```diff
@@ -1,15 +1,23 @@
 import { Agent } from "@aries-framework/core";
 import { agentModules } from "../agent";
 import { BasicStoragesApi } from "../../storage/storage.types";
+import { SignifyApi } from "../modules/signify/signifyApi";
 
 abstract class AgentService {
   protected readonly agent: Agent<typeof agentModules>;
 
-  protected readonly basicStorage!: BasicStoragesApi;
+  protected readonly basicStorage: BasicStoragesApi;
 
-  constructor(agent: Agent, basicStorage: BasicStoragesApi) {
+  protected readonly signifyApi: SignifyApi;
+
+  constructor(
+    agent: Agent,
+    basicStorage: BasicStoragesApi,
+    signifyApi: SignifyApi
+  ) {
     this.agent = agent;
-    this.basicStorage = basicStorage!;
+    this.basicStorage = basicStorage;
+    this.signifyApi = signifyApi;
```

Comment:
Same comment as in the other PR - can we just keep this as a getter of the agent, and call `this.agent.signify`?

Reply:
based on convo on slack - lets leave this too, my misunderstanding